Library/DasharoSystemFeaturesUiLib/DasharoSystemFeaturesVfr.vfr

```diff
@@ -232,13 +232,25 @@ formset
          endoneof;
     endif;
 
-    oneof varid = FeaturesData.SleepType,
-             prompt = STRING_TOKEN(STR_SLEEP_TYPE_PROMPT),
-             help   = STRING_TOKEN(STR_SLEEP_TYPE_HELP),
+    suppressif ideqval FeaturesData.PowerMenuShowSleepType == 0;
+         oneof varid = FeaturesData.SleepType,
+                  prompt = STRING_TOKEN(STR_SLEEP_TYPE_PROMPT),
+                  help   = STRING_TOKEN(STR_SLEEP_TYPE_HELP),
+                  flags  = RESET_REQUIRED,
+
+                  option text = STRING_TOKEN(STR_SLEEP_TYPE_S0IX), value = SLEEP_TYPE_S0IX, flags = DEFAULT;
+                  option text = STRING_TOKEN(STR_SLEEP_TYPE_S3), value = SLEEP_TYPE_S3, flags = 0;
+         endoneof;
+    endif;
+
+    oneof varid = FeaturesData.PowerFailureState,
+             prompt = STRING_TOKEN(STR_POWER_FAILURE_STATE_PROMPT),
+             help   = STRING_TOKEN(STR_POWER_FAILURE_STATE_HELP),
              flags  = RESET_REQUIRED,
 
-             option text = STRING_TOKEN(STR_SLEEP_TYPE_S0IX), value = SLEEP_TYPE_S0IX, flags = DEFAULT;
-             option text = STRING_TOKEN(STR_SLEEP_TYPE_S3), value = SLEEP_TYPE_S3, flags = 0;
+             option text = STRING_TOKEN(STR_POWER_FAILURE_STATE_OFF), value = POWER_FAILURE_STATE_OFF, flags = DEFAULT;

```

Comment:
```suggestion
             option text = STRING_TOKEN(STR_POWER_FAILURE_STATE_OFF), value = POWER_FAILURE_STATE_OFF, flags = 0;
```

The default value should be equal to PcdDefaultPowerFailureState passed from coreboot. You should add a default restoration logic to the DasharoSystemFeaturesCallback @SergiiDmytruk

Reply:
Right, I didn't try to restore default.  By the way, shouldn't `*_QUESTION_ID` macros be used in code and VFR?

Reply:
Feel free to patch things if you know what is missing @SergiiDmytruk

Reply:
OK, I asked because maybe it wasn't done intentionally for some non-obvious reason.